packages/docs/concepts/references-and-templates.yaml

```diff
@@ -0,0 +1,224 @@
+        type: MarkdownWithCode
+        properties:
+          content: |
+            References are used to split the configuration of an app into logically distinct files, and to reuse configuration in the app. References can be used almost anywhere in the configuration, as long as the configuration remains valid YAML.
+
+            References are made using the [`_ref`](/_ref) operator. If the referenced file has a `.yaml` or `.json` extension, the contents of the file will be parsed, else the file content is included as a string (this is useful for `.md` or `.html` files). As an example of splitting an app into logically distinct files, references can be used to write each page as a separate file:
+
+            ###### lowdefy.yaml
+            ```yaml
+            lowdefy: LOWDEFY_VERSION
+
+            pages:
+              _ref: pages/page1.yaml # Path to the referenced file. Always from the root of the project.
+              _ref: pages/page1.yaml
+            ```
+
+            The `_ref` operator can take an argument called `vars`. This can be any data, and is passed down to later be accessed with the [`_var`](/_var) operator. By using vars, the referenced file can become a template, using the given variables. For example, a standard page template might be used for multiple pages in an app:
+
+            ###### pages/page1.yaml
+            ```yaml
+            _ref:
+              path: templates/text_page.yaml
+              vars:
+                id: page1
+                title: Page 1
+                content:
+                  _ref: markdowns/content.md
+            ```
+
+            ###### markdowns/content.md
+            ```yaml
+            Page content text.
+            ```
+
+            ###### templates/text_page.yaml
+            ```yaml
+            id:
+              _var: id
+            type: PageHeaderMenu
+            properties:
+              title:
+                _var: title
+            blocks:
+              - id: content_card
+                type: Card
+                blocks:
+                  - id: title
+                    type: Title
+                    properties:
+                      content:
+                        _var: title
+                  - id: content
+                    type: Markdown
+                    properties:
+                      content:
+                        _var: content
+            ```
+
+            ## Nunjucks Templating
+
+            Templating can be taken further by referencing [Nunjucks](https://mozilla.github.io/nunjucks/) template files. If a file ends with the `.njk` file extension, the file will first be hydrated as a Nunjucks template, using the `vars` as template variables. If the file ends with `.yaml.njk` or `.json.njk`, the output of the template will then be parsed. Nunjucks templates are useful since the template file does not need to be valid yaml before it is hydrated, and features like for-loops and if-statements can be used.
```

Comment:
> Nunjucks templates are useful since the template file does not need to be valid yaml before it is hydrated, and features like for-loops and if-statements can be used.

I would not say thats the reason why they are useful. Just also update this sentance from previous changed above.

Reply:
Change made in 9468acfa6e458bb72216e485b520fa0cf37d8cb4